hive/agents/qnets/utils.py

```diff
@@ -20,9 +20,12 @@ def calculate_output_dim(net, input_shape):
     """
     if isinstance(input_shape, int):
         input_shape = (input_shape,)
-    placeholder = torch.zeros((0,) + tuple(input_shape))
+    placeholder = torch.zeros((1,) + tuple(input_shape)).to(device)
     output = net(placeholder)
-    return output.size()[1:]
+    if isinstance(output, tuple):
```

Comment:
If it's a tuple, you should return the size of each output. It doesn't make sense to only do the first output. That's just specific to your current use case

Reply:
Probably should be recursive? Keep going until you hit scalars or tensors